odoo/addons/base/models/ir_actions_report.py

```diff
@@ -45,8 +46,18 @@
 # before rendering a barcode (done in a C extension) and this part is not thread safe. We attempt
 # here to init the T1 fonts cache at the start-up of Odoo so that rendering of barcode in multiple
 # thread does not lock the server.
+_DEFAULT_BARCODE_FONT = 'Courier'
 try:
-    createBarcodeDrawing('Code128', value='foo', format='png', width=100, height=100, humanReadable=1).asString('png')
+    available = TypeFace(_DEFAULT_BARCODE_FONT).findT1File()
+    if not available:
+        _DEFAULT_BARCODE_FONT = 'NimbusMonoPS-Regular'
+    createBarcodeDrawing('Code128', value='foo', format='pdf', width=100, height=100, humanReadable=1, fontName=_DEFAULT_BARCODE_FONT).asString('pdf')
+    if not available:
```

Comment:
why repeat the same condition as above ? You could set the ascent and descent in the first `if` I think

Reply:
right, wrong reminder of previous tests
fixed, now _DEFAULT_BARCODE_FONT is either Courier or the substitution font if found